PowerEditor/src/Parameters.cpp

```diff
@@ -2244,6 +2247,113 @@ void NppParameters::updateLangXml(TiXmlElement* mainElemUser, TiXmlElement* main
 
 void NppParameters::updateStylesXml(TiXmlElement* rootUser, TiXmlElement* rootModel, TiXmlElement* mainElemUser, TiXmlElement* mainElemModel)
 {
+	std::wstring defaultFgColor, defaultBgColor;
+
+	auto endsWith = [](std::wstring const& fullString, std::wstring const& suffix) -> bool
+	{
+		if (fullString.length() >= suffix.length()) {
+			// Compare the last 'suffix.length()' characters of 'fullString' with 'suffix'
+			return (0 == fullString.compare(fullString.length() - suffix.length(), suffix.length(), suffix));
+		}
+		else {
+			return false;
+		}
+	};
+	bool useDefaultColors = !endsWith(rootUser->GetDocument()->Value(), L"stylers.xml");	// use the Colors from "Default Style", except when it's stylers.xml
+
+	// Start with GlobalStyles
+	//		(even though it comes later in the actual XML file, need to be able to extract the defaultFgColor and defaultBgColor before doing the individual lexers)
+	TiXmlNode* gsUser = rootUser->FirstChildElement(L"GlobalStyles");
+	TiXmlNode* gsModel = rootModel->FirstChildElement(L"GlobalStyles");
+	if (!gsUser || !gsModel)
+		return;
+
+	// map UserStyler's widget styleID||name -> node-pointer
+	std::map<std::wstring, TiXmlElement*> mapUserWidgets{};
+	for (TiXmlElement* widgetFromUser = gsUser->FirstChildElement(L"WidgetStyle");
+		widgetFromUser;
+		widgetFromUser = widgetFromUser->NextSiblingElement(L"WidgetStyle"))
+	{
+		// use StyleID for the map's key, or if styleID not found or if "0" then use the widget's name (lowercase) instead
+		std::wstring widgetKey = widgetFromUser->Attribute(L"styleID");
+		if (widgetKey.empty() || widgetKey == L"0" || (decStrVal(widgetKey.c_str()) > 256) || (decStrVal(widgetKey.c_str()) < 0))
+			widgetKey = widgetFromUser->Attribute(L"name");
+
+		// add widget to map using the key
+		if (widgetKey.length())
+		{
+			widgetKey = stringToLower(widgetKey);

```

Comment:
Why "stringToLower"?

Reply:
Case is not always identical in the name.  I ran across it in the debugging, so included the `stringToLower`:

For example:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/bd3988817ba44e7bc94b88faa9151a4cef0ea9e6/PowerEditor/installer/themes/DarkModeDefault.xml#L1804
vs
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/bd3988817ba44e7bc94b88faa9151a4cef0ea9e6/PowerEditor/installer/themes/DansLeRuSH-Dark.xml#L1186

Reply:
Also, I re-ran without the `stringToLower` calls, and it is working correctly still -- I think I just must've been seeing a problem when I didn't have all the logic correct/debugged, and the stringToLower wasn't needed long-term (because of using the styleID as the widgetKey for numbered styles).  So I have removed the stringToLower calls.